pelican/__init__.py

```diff
@@ -80,7 +80,8 @@ def init_plugins(self):
                 plugin.register()
                 self.plugins.append(plugin)
             except Exception as e:
-                logger.error("Cannot register plugin `%s`\n%s", name, e)
+                logger.error("Cannot register plugin `%s`\n%s", name, e, stacklevel=3)
+                print(e.stacktrace)
```

Comment:
No to `print`. Pass `exc_info` to `logger.error` ([example](https://github.com/getpelican/pelican/blob/5d3e87b50be0c84d272e7fb2f9fb794af7887b50/pelican/generators.py#L873C27-L878))

Reply:
So I dropped `print()`, and instead am conditionally using Rich's `console.print_exception()`. This provides a fair bit more context to what is happening. Is this an okay way to do it?

Using `exc_info`, you get something like this:

```
           ERROR    Cannot register plugin `minchin.pelican.readers.commonmark`                           __init__.py:66
                    module 'pelican.plugins.signals' has no attribute 'make_something_up'
                    Traceback (most recent call last):
                      File "D:\Code\pelican\pelican\__init__.py", line 80, in init_plugins
                        plugin.register()
                      File
                    "D:\Code\minchin.pelican.readers.commonmark\minchin\pelican\readers\commonmark\__init
                    __.py", line 19, in register
                        signals.make_something_up.connect(check_settings())
                        ^^^^^^^^^^^^^^^^^^^^^^^^^
                    AttributeError: module 'pelican.plugins.signals' has no attribute 'make_something_up'
```

With `console.print_exception()`, you get something like this:

```
           ERROR    Cannot register plugin `minchin.pelican.readers.commonmark`                           __init__.py:66
                    module 'pelican.plugins.signals' has no attribute 'make_something_up'
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ D:\Code\pelican\pelican\__init__.py:80 in init_plugins                                           │
│                                                                                                  │
│    77 │   │   │   name = get_plugin_name(plugin)                                                 │
│    78 │   │   │   logger.debug("Registering plugin `%s`", name)                                  │
│    79 │   │   │   try:                                                                           │
│ ❱  80 │   │   │   │   plugin.register()                                                          │
│    81 │   │   │   │   self.plugins.append(plugin)                                                │
│    82 │   │   │   except Exception as e:                                                         │
│    83 │   │   │   │   logger.error(                                                              │
│                                                                                                  │
│ D:\Code\minchin.pelican.readers.commonmark\minchin\pelican\readers\commonmark\__init__.py:19 in  │
│ register                                                                                         │
│                                                                                                  │
│   16 │   signals.initialized.connect(check_settings)                                             │
│   17 │   signals.initialized.connect(commonmark_version)                                         │
│   18 │   signals.readers_init.connect(add_commonmark_reader)                                     │
│ ❱ 19 │   signals.make_something_up.connect(check_settings())                                     │
│   20                                                                                             │
│   21                                                                                             │
│   22 from .constants import LOG_PREFIX                                                           │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
AttributeError: module 'pelican.plugins.signals' has no attribute 'make_something_up'
```

(both in colour)

Reply:
> Is this an okay way to do it?

Yes-ish... Technically, both are more or less the same thing, but now it'll bug me that this is done differently in different places :). No big deal though.